src/task.ts

```diff
@@ -377,6 +379,107 @@ export class WhenAnyTask extends CompoundTask {
     }
 }
 
+/**
+ * @hidden
+ *
+ * A long Timer Task.
+ *
+ * This Task is created when a timer is created with a duration
+ * longer than the maximum timer duration supported by storage infrastructure.
+ *
+ * It extends `WhenAllTask` because it decomposes into
+ * several smaller sub-`TimerTask`s
+ */
+export class LongTimerTask extends WhenAllTask implements TimerTask {
+    public id: TaskID;
+    public action: CreateTimerAction;
+    private readonly executor: TaskOrchestrationExecutor;
+    private readonly maximumTimerLength: moment.Duration;
+    private readonly orchestrationContext: DurableOrchestrationContext;
+    private readonly longRunningTimerIntervalDuration: moment.Duration;
+
+    public constructor(
+        id: TaskID,
+        action: CreateTimerAction,
+        orchestrationContext: DurableOrchestrationContext,
+        executor: TaskOrchestrationExecutor,
+        maximumTimerLength: moment.Duration,
+        longRunningTimerIntervalDuration: moment.Duration
+    ) {
+        const currentTime = moment(orchestrationContext.currentUtcDateTime);
+        const finalFireTime = moment(action.fireAt);
+        const durationTillFire = moment.duration(finalFireTime.diff(currentTime));
```

Comment:
nit: just for style's sake, can we change `durationTillFire` to `durationUntilFire`? Just trying to avoid the abbreviations :)

Alternatively, I also like `timerLength` for this variable.

Reply:
Done, removed the abbreviation :)